src/explorer/connectionTreeItem.ts

```diff
@@ -53,58 +60,65 @@ export default class ConnectionTreeItem extends vscode.TreeItem implements TreeI
   }
 
   getChildren(): Thenable<any[]> {
-    if (this.isExpanded) {
-      if (this._childrenCacheIsUpToDate) {
-        return Promise.resolve(this._childrenCache);
-      } else {
-        return new Promise(async (resolve, reject) => {
-          // If we aren't the active connection, we reconnect.
-          if (this._connectionController.getActiveConnectionInstanceId() !== this._connectionInstanceId) {
-            try {
-              await this._connectionController.connectWithInstanceId(this._connectionInstanceId);
-            } catch (err) {
-              return reject(err);
-            }
+    if (!this.isExpanded || this._connectionController.isDisconnecting() || this._connectionController.isConnnecting()) {
+      return Promise.resolve([]);
+    }
+
+    if (this._childrenCacheIsUpToDate) {
+      return Promise.resolve(Object.values(this._childrenCache));
+    } else {
+      return new Promise(async (resolve, reject) => {
```

Comment:
I have seen this `new Promise(async (resolve, reject))` syntax in master and eslint didn't like it :) so I switched off `no-misused-promises` rule.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-misused-promises.md

But maybe we can avoid this syntax in the future and in this particular case you removed `await this._connectionController`, so you maybe don't need `async` keyword at all.